ipam/manager_test.go

```diff
@@ -141,233 +141,279 @@ func cleanupTestAddressSpace(am AddressManager) error {
-	}
-
-	err = am.ReleasePool(LocalDefaultAddressSpaceId, poolId)
-	if err != nil {
-		t.Errorf("ReleasePool failed, err:%v", err)
-	}
-}
+var (
+	_ = Describe("Test Manager", func() {
+
+		Describe("Test Initialize", func() {
+			Context("When store is nil", func() {
+				It("Initialize return nil", func() {
+					var config common.PluginConfig
+					config.Store = nil
+					options := map[string]interface{}{}
+					options[common.OptEnvironment] = ""
+					am, err := NewAddressManager()
+					Expect(am).NotTo(BeNil())
+					Expect(err).NotTo(HaveOccurred())
+					err = am.Initialize(&config, options)
+					Expect(err).To(BeNil())
+				})
+			})
+
+			Context("When restore key not found", func() {
+				It("Initialize return nil", func() {
+					var config common.PluginConfig
+					storeMock := &testutils.KeyValueStoreMock{}
+					storeMock.ReadError = store.ErrKeyNotFound
+					config.Store = storeMock
+					options := map[string]interface{}{}
+					options[common.OptEnvironment] = ""
+					am, err := NewAddressManager()
+					Expect(am).NotTo(BeNil())
+					Expect(err).NotTo(HaveOccurred())
+					err = am.Initialize(&config, options)
+					Expect(err).To(BeNil())
+				})
+			})
+
+			Context("When restore return error", func() {
+				It("Initialize return error", func() {
+					var config common.PluginConfig
+					storeMock := &testutils.KeyValueStoreMock{}
+					storeMock.ReadError = errors.New("Error")
+					config.Store = storeMock
+					options := map[string]interface{}{}
+					options[common.OptEnvironment] = ""
+					am, err := NewAddressManager()
+					Expect(am).NotTo(BeNil())
+					Expect(err).NotTo(HaveOccurred())
+					err = am.Initialize(&config, options)
+					Expect(err).To(HaveOccurred())
+				})
+			})
+
+			Context("When StartSource fail", func() {
+				It("Initialize return error", func() {
+					var config common.PluginConfig
+					options := map[string]interface{}{}
+					options[common.OptEnvironment] = "Invalid"
+					am, err := NewAddressManager()
+					Expect(am).NotTo(BeNil())
+					Expect(err).NotTo(HaveOccurred())
+					err = am.Initialize(&config, options)
+					Expect(err).To(HaveOccurred())
+				})
+			})
+		})
+
+		Describe("Test restore", func() {
+			Context("When store is nil", func() {
+				It("restore return nil", func() {
+					am := &addressManager{
+						AddrSpaces: make(map[string]*addressSpace),
+					}
+					err := am.restore()
+					Expect(err).To(BeNil())
+				})
+			})
+
+			Context("Test Populate pointers", func() {
+				It("Should build addrsByID successfully", func() {
+					am := &addressManager{
+						AddrSpaces: make(map[string]*addressSpace),
+					}
+					timeReboot, _ := platform.GetLastRebootTime()
+					am.store = &testutils.KeyValueStoreMock{
+						ModificationTime: timeReboot.Add(time.Hour),
+					}
+					ap := &addressPool{
+						Id:			"ap-test",
+						RefCount: 	1,
+						Addresses: 	make(map[string]*addressRecord),
+					}
+					ap.Addresses["ar-test"] = &addressRecord{
```

Comment:
can we also add some addr and check if addrbyid returns that address...thats the main purpose of addrsbyid